packages/common/src/utils/dependencies.ts

```diff
@@ -42,10 +58,14 @@ export async function fetchPackageJSON(dep: string, version: string) {
     // If it is not an absolute version (e.g. a tag like `next`), we don't want to fetch
     // using JSDelivr, because JSDelivr caches the response for a long time. Because of this,
     // when a tag updates to a new version, people won't see that update for a long time.
-    // Unpkg does handle this nicely, but is less stable. So we default to JSDelivr, but
-    // for tags we use unpkg.
+    // Instead, we download all possible versions from JSDelivr, and we check those versions
+    // to see what's the maximum satisfying version. The API call is cached for only 10s.
     try {
-      return await fetchUnpkg();
+      const allVersions = await fetchAllVersions(dep);
+      return (
+        allVersions.tags[version] ||
+        maxSatisfying(allVersions.versions, version)
+      );
     } catch (e) {
       return fetchJsdelivr();
```

Comment:
Shouldn't it just fail in this case as the try is also jsdelivr?

Reply:
OH BOY, I merged before, apologies! I think this is actually better to fall back to unpkg yes...

Reply:
Will add that as additional commit./